src/bindings/c/ov/include/c_api/ov_c_api.h

```diff
@@ -0,0 +1,1122 @@
+OPENVINO_C_API(void) ov_model_free(ov_model_t *model);
+
+/**
+ * @brief Release the memory allocated by ov_compiled_model_t.
+ * @param compiled_model A pointer to the ov_compiled_model_t to free memory.
+ */
+OPENVINO_C_API(void) ov_compiled_model_free(ov_compiled_model_t *compiled_model);
+
+/**
+ * @brief Creates a compiled model from a source model object.
+ * Users can create as many compiled models as they need and use
+ * them simultaneously (up to the limitation of the hardware resources).
+ * @param core A pointer to the ie_core_t instance.
+ * @param model Model object acquired from Core::read_model.
+ * @param device_name Name of a device to load a model to.
+ * @param compiled_model A pointer to the newly created compiled_model.
+ * @param property Optional pack of pairs: (property name, property value) relevant only for this load operation
+ * operation.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_core_compile_model(const ov_core_t* core,
+                                            const ov_model_t* model,
+                                            const char* device_name,
+                                            ov_compiled_model_t **compiled_model,
+                                            const ov_property_t* property);
+
+/**
+ * @brief Reads a model and creates a compiled model from the IR/ONNX/PDPD file.
+ * This can be more efficient than using the ov_core_read_model_from_XXX + ov_core_compile_model flow,
+ * especially for cases when caching is enabled and a cached model is available.
+ * @param core A pointer to the ie_core_t instance.
+ * @param model_path Path to a model.
+ * @param device_name Name of a device to load a model to.
+ * @param compiled_model A pointer to the newly created compiled_model.
+ * @param property Optional pack of pairs: (property name, property value) relevant only for this load operation
+ * operation.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_core_compile_model_from_file(const ov_core_t* core,
+                                                    const char* model_path,
+                                                    const char* device_name,
+                                                    ov_compiled_model_t **compiled_model,
+                                                    const ov_property_t* property);
+                                                    
+/**
+ * @brief Sets properties for a device, acceptable keys can be found in ov_property_key_e.
+ * @param core A pointer to the ie_core_t instance.
+ * @param device_name Name of a device.
+ * @param property ov_property propertys.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_core_set_property(const ov_core_t* core,
+                                        const char* device_name,
+                                        const ov_property_t* property);
```

Comment:
can we use this method with device name as NULL?
We need a  way to set COre properties like cache_dir, tbb_terminate